src/modules/organization/event/components/table/use-table/index.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ReviewJoinRequestInput,
+  UserEventStatus
+} from "../../../../../../apollo/types";
+import { useReviewRequest } from "../../../use-org-event/api";
+import { DataProps } from "../types";
+import { ref, computed, Ref } from "vue";
+import Fuse from "fuse.js";
+import { useRoute } from "vue-router";
+
+const useTable = (data?: Ref<DataProps[] | undefined>) => {
+  const searchValue = ref("");
+  const sortOption = ref("descending");
+  const handleSearch = (value: string) => {
+    sortOption.value = "default";
+    searchValue.value = value;
+  };
+  const sortBy = ["Descending alphabets", "Ascending alphabets"];
+  const sortByVal = ["descend", "ascend"];
+  const header: Record<string, string> = {
+    name: "Name",
+    email: "Email",
+    phoneNumber: "Phone",
+    status: "Status"
+  };
+  const headerKeys = computed(() => {
+    return Object.keys(header);
+  });
+  const filteredData = computed(() => {
+    if (!data?.value) return [];
+
+    const fuse = new Fuse(data.value, {
```

Comment:
can we move fuse outside of this scope. by declaring fuse inside of computed of `searchValue` It will initialize fuse everytime we change searchValue without necessary

Recommended way
```
const fuse = computed(() => {
     //declare fuse here
     return new Fuse()
})

const filteredData = computed(() => {
    // use fuse here
})
```